src/modules/blockchainApplication/manage/hooks/useCurrentApplication.js

```diff
@@ -1,18 +1,48 @@
 import { useCallback } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import client from 'src/utils/api/client';
+import { selectStaking } from 'src/redux/selectors';
+import { stakesReset } from 'src/redux/actions';
+import {
+  removeThenAppendSearchParamsToUrl,
+  removeSearchParamsFromUrl,
+} from 'src/utils/searchParams';
 import { selectCurrentApplication } from '../store/selectors';
 import { setCurrentApplication } from '../store/action';
 
-export function useCurrentApplication() {
+export function useCurrentApplication(history) {
```

Comment:
Why did we change from using window? Or our own util `import history from 'src/utils/history';` for this?

Dont we now we need to pass in history from all places that uses `setApplication`, for example in `SelectNode`, `useApplicationManagement`, etc..

Reply:
@oskarleonard Using the passed in history lets us set the state for the confirmation dialog via the location state rather than via the URL. When we upgrade the react-router packages then we can use the useHistory hook instead

Reply:
But than we need to pass in the history in many other places, my second paragraph?

If that is a big refactor we could do

```
import history from 'src/utils/history';
export function useCurrentApplication(history) {
removeThenAppendSearchParamsToUrl(history, { modal: 'confirmationDialog' }, ['modal'], state);
```
And in ConfirmationDialog.js
```
import history from 'src/utils/history';
function ConfirmationDialog() {
const { header, content, cancelText, onCancel, confirmText, onConfirm } = history.location.state ?? {};
```

Reply:
It's not a big refactor to pass in history because we only actively try to switch applications via the application/network dialog. While bootstrapping the application for instance, staking queue will be empty and hence no need to worry about it. `history.location` is mutable so it's never a good idea to use it. Then the history import you specified is used in only two places excluding tests. Therefore I chose not to use it because I'm unaware of whatever challenges have been had with it. The best route is still the react-router package update and I've added a TODO for that.